UNO-628 breadcrumbs

left23 opened the pull request:
Two modules to install (run `composer install`)

- https://www.drupal.org/project/menu_position
- https://www.drupal.org/project/menu_breadcrumb

Config to import (`drush cim`) for above modules, to place breadcrumb block, and to create 2 news path alias patterns.
Bulk regenerate and/or delete then regenerate path aliases for Content (Story, Event) if there are existing nodes.
There is also config changes to update paths for Stories and Events Views pages.

See UNO-628 for more info.

left23 commented:
@orakili For review at your leisure. I don't want to merge until I get a second opinion but it's not blocking anything. I will deploy as a feature branch to the dev site to test

orakili commented:
@left23 I tested locally and I could see the breadcrumbs above the title:

<img width="520" alt="Screenshot 2023-05-10 at 13 24 21" src="https://github.com/UN-OCHA/unocha-site/assets/696348/4a4c8de3-f531-4ac6-8273-e8b20b155492">

But I got the same thing when disabling the `menu_position` module so I'm wondering what it is used for. Could you elaborate?

left23 commented:
@orakili Sure, the menu_position module is there to help the breadcrumbs on pages like Story and Event nodes which are not in the menu, and thus the menu_breadcrumb doesn't work for these cases.
I think if you were to disable the menu_position module and visit a Story, the Story nodes (see https://new.dev.unocha-org.ahconu.org/latest/news-and-stories/7-reasons-why-world-must-keep-supporting-ukraine) breadcrumbs wouldn't appear as they do on the dev site, but instead like: `Home/News and Stories`, without reflecting the correct and entire menu structure in the breadcrumbs.

left23 commented:
I've made some updates to the path alias for Story and Event content type, so we no longer have the extra word "latest" in the URL, although this remains in the breadcrumb because it reflects the Menu structure.
So for Events and Story content types, their paths are `/events/[node-title]`, `/news/[node-title]`

I've also added a menu_position item for Publications so instead of having the breadcrumb be: `Home/Publications` as is currently on the dev site https://new.dev.unocha-org.ahconu.org/publications/report/mali/rapport-forum-national-sur-la-localisation-de-laide-au-mali-fona-lahdm-1ere-edition, 
![Selection_098](https://github.com/UN-OCHA/unocha-site/assets/1835923/3d57aa09-77dd-4898-9546-2afc73106294)

we have the extra word "Latest" (which is the parent menu item) appearing in the breadcrumbs.
![Selection_095](https://github.com/UN-OCHA/unocha-site/assets/1835923/7303c5e8-16ef-4a1c-83f2-b2ffa89c4f4d)
This has the added benefit of making the Main menu parent items active (Latest > Publications) so we have a visual queue in the main menu that reflects the active trail.

Lastly, I've added a CSS rule to hide the page title (last item) in the breadcrumbs for Story, Event content types and for Publication items. 
![Selection_096](https://github.com/UN-OCHA/unocha-site/assets/1835923/959a95a7-2739-4a3f-93c0-6a4832a8190e)

For Basic, Response and Region node types, the page title still displays in the breadcrumb
![Selection_097](https://github.com/UN-OCHA/unocha-site/assets/1835923/f5fab3eb-710d-4cc1-b8ae-58f656ae14ec)

left23 commented:
@orakili In order to help deploy recent updates from the `develop` branch, I'm disabling the `menu_position` module here as we intended and deploying `feature/UNO-628-breadcrumbs` branch to dev site with latest dev updates. `menu_position` will still need to be remove via composer.
